toolchains/sysimage/toolchain.bzl

```diff
@@ -395,74 +390,6 @@ ext4_image = _icos_build_rule(
     },
 )
 
-def _inject_files_impl(ctx):
-    tool = ctx.files._inject_files[0]
-    dflate = ctx.files._dflate[0]
-
-    out = ctx.actions.declare_file(ctx.label.name)
-
-    inputs = [ctx.files.base[0]]
-
-    args = [
-        "--input",
-        ctx.files.base[0].path,
-        "--output",
-        out.path,
-        "--dflate",
-        dflate.path,
-    ]
-
-    if len(ctx.files.file_contexts) > 0:
-        args += ["--file-contexts", ctx.files.file_contexts[0].path]
-        inputs += ctx.files.file_contexts
-
-    if ctx.attr.prefix:
-        args += ["--prefix", ctx.attr.prefix]
-
-    for input_target, install_target in ctx.attr.extra_files.items():
-        args.append(input_target.files.to_list()[0].path + ":" + install_target)
-        inputs += input_target.files.to_list()
-
-    _run_with_icos_wrapper(
-        ctx,
-        executable = tool.path,
-        arguments = args,
-        inputs = inputs,
-        outputs = [out],
-        tools = [tool, dflate],
-    )
-
-    return [DefaultInfo(files = depset([out]))]
-
-inject_files = _icos_build_rule(
```

Comment:
I've removed this rule, but the binary is still in the repo [here](https://github.com/dfinity/ic/tree/55fc36e3fb4c6eeb39b9a59e1655b61465fb5a5c/rs/ic_os/build_tools/inject_files). Neither are in use, but I think the Rust is easier to maintain and could be useful later.